src/components/layout/nav/Navbar.jsx

```diff
@@ -1,7 +1,112 @@
-import React from "react";
+import React, { useReducer } from "react";
+import Container from "react-bootstrap/Container";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+import { NavLink } from "react-router-dom";
+import SignUpBox from "./signup/SignUp";
+import SearchBox from "./search/Search";
+import "../../home/HomePage.scss";
+
+function navbarIconsReducer(state, action) {
+  switch (action.type) {
+    case "IS_SIGN_OPENED":
+      return {
+        ...state,
+        isSignBoxOpen: !state.isSignBoxOpen,
+        isSearchBoxOpen: false,
+      };
+
+    case "IS_SEARCH_OPENED":
+      return {
+        ...state,
+        isSearchBoxOpen: !state.isSearchBoxOpen,
+        isSignBoxOpen: false,
+      };
+
+    case "IS_HAMBURGER_OPENED":
+      return {
+        isHamburgerOpen: !state.isHamburgerOpen,
+        isSignBoxOpen: false,
+        isSearchBoxOpen: false,
+      };
+
+    default:
+      return state;
+  }
+}
 
 const Navbar = () => {
-  return <div></div>;
+  const [
+    { isSignBoxOpen, isSearchBoxOpen, isHamburgerOpen },
+    dispatch,
+  ] = useReducer(navbarIconsReducer, {
+    isSignBoxOpen: false,
+    isSearchBoxOpen: false,
+    isHamburgerOpen: false,
+  });
+
+  const handleStatusOfIcons = (type) => dispatch({ type });
+
+  return (
+    <>
+      {isSearchBoxOpen && <SearchBox />}
+      <Container fluid className="navbar">
+        <Row
+          className="hamburgerContainer navbarItemWrapper"
+          onClick={() => handleStatusOfIcons("IS_HAMBURGER_OPENED")}
+        >
+          <Col className="hamburgerIcon"></Col>
+        </Row>
+        <Row
```

Comment:
to make this code more readable, i think you should split out the hamburger into a variable
```
const hamburgerMenu = (<Row className="hamburgerContainer" ...)
const fullScreeNavMenu = (...)

<Container>
{hamburgerMenu}
{fullScreenNavMenu}
</Container>
```